go/vt/vtgate/vtgate.go

```diff
@@ -303,7 +306,7 @@ func Init(
 	}
 
 	initAPI(gw.hc)
-
+	planbuilder.EnableShardRoutingFlag(*EnableShardRouting)
```

Comment:
I think we should not set this like this.
instead in planbuilder we can call into vcusor to check if the flag is enabled or disabled.